### DISCUSSION TITLE
ios not workink on long dynamic link

### DISCUSSION THREAD

**Discussion by bcg-dev** [invertase/react-native-firebase#6219] [Q&A]
The short dynamic link works well, but long dynamic link not working properly.
Those links from firebase console.

Long dynamic link only move to my app not fired dynamic link function. (etc. getInitialLink, onLink)
And on android, both works perfect.
Could anyone tell me why? please save my days

**short dynamic link : https://soyeon.page.link/XktS
long dynamic link : https://soyeon.page.link/?link=https://www.soyeon.kr/&apn=com.soyeon.soyeon2&isi=1610955216&ibi=com.soyeon.soyeon**
(replaced company name to my name)

here is App.js
```
const getDynamicInitialLink = useCallback(async () => {
        const initialLink = await dynamicLinks().getInitialLink();

        if (initialLink) {
            commonHandleLink(initialLink);
        }
    }, []);

const handleDynamicLink = link => {
        commonHandleLink(link);
    }

    const commonHandleLink = link => {
        if(link.url.includes("parameter")){
            .......
        }
    }

useEffect(() => {
        console.disableYellowBox = true;

        getDynamicInitialLink();


        AppState.addEventListener("change", handleAppStateChange);
        const dynamicLink = dynamicLinks().onLink(handleDynamicLink);

        return () => {
            dynamicLink();
        }
    }, [webViewRef]);
```

thanks in advance and sorry my english.

**Comment by mikehardy** [comment]
Hmm - it might be that the long dynamic link does not look enough like a dynamic link to trigger it. The strategy I use (and that most people use I think?) is to also listen to the react-native default getInitialLink method as well, and inspect it to see if it is a dynamic link.

Perhaps if you gave us more details on your implementation (like the actual links examples and an App.js showing how to reproduce it) we could help more

https://stackoverflow.com/help/minimal-reproducible-example / https://stackoverflow.com/help/how-to-ask

**Reply by bcg-dev** [reply]
Okay I was in a hurry. I edited my question !

**Comment by carlosmoliv** [comment]
Hey, I'm having the same problem, any updates @bcg-dev?

**Comment by MiteshKalal7** [comment]
Did you find the way?
